src/main/java/de/tum/in/www1/artemis/service/LectureUnitService.java

```diff
@@ -70,6 +65,49 @@ public void setLectureUnitCompletion(@NotNull LectureUnit lectureUnit, @NotNull
         }
     }
 
+    /**
+     * Set the completion status of all passed lecture units for the give user
+     * If the user completed the unit and completion status already exists, nothing happens
+     *
+     * @param lectureUnits List of all lecture units for which to set the completion flag
+     * @param user         The user that completed/uncompleted the lecture unit
+     * @param completed    True if the lecture unit was completed, false otherwise
+     */
+    public void completeAllLectureUnits(List<? extends LectureUnit> lectureUnits, @NotNull User user, boolean completed) {
```

Comment:
Right now the method name suggest that you set complete = true for all units
```suggestion
    public void setCompletedForAllLectureUnits(List<? extends LectureUnit> lectureUnits, @NotNull User user, boolean completed) {
```